include/xtensor/xassign.hpp

```diff
@@ -457,6 +509,303 @@ namespace xt
+
+        auto fct_stepper = e2.stepper_begin(e1.shape());
+        auto res_stepper = e1.stepper_begin(e1.shape());
+    
+        // TODO in 1D case this is ambigous -- could be RM or CM. Use default layout to make decision
+        if (e1.strides().back() == 1) // row major case
+        {
+            for (std::size_t ox = 0; ox < outer_loop_size; ++ox)
+            {
+                for (std::size_t i = 0; i < simd_size; i++)
+                {
+                    res_stepper.template store_simd<simd_type>(fct_stepper.template step_simd<simd_type>());
+                }
+                for (std::size_t i = 0; i < simd_rest; ++i)
+                {
+                    *(res_stepper) = *(fct_stepper);
+                    res_stepper.step_leading();
+                    fct_stepper.step_leading();
+
+                }
+                strided_assign_detail::idx_tools<layout_type::row_major>::next_idx(idx, max);
+
+                fct_stepper.to_begin();
+                // need to step E1 as well if not contigous assign (e.g. view)
+                if (!E1::contiguous_layout)
+                {
+                    res_stepper.to_begin();
+                    for (std::size_t i = 0; i < idx.size(); ++i)
+                    {
+                        fct_stepper.step(i, idx[i]);
+                        res_stepper.step(i, idx[i]);
+                    }
+                }
+                else
+                {
+                    for (std::size_t i = 0; i < idx.size(); ++i)
+                    {
+                        fct_stepper.step(i, idx[i]);
+                    }
+                }
+            }
+        }
+        else if(e1.strides().front() == 1)
+        {
+            for (std::size_t ox = 0; ox < outer_loop_size; ++ox)
+            {
+                for (std::size_t i = 0; i < simd_size; i++)
+                {
+                    res_stepper.template store_simd<simd_type>(fct_stepper.template step_simd<simd_type>());
+                }
+                for (std::size_t i = 0; i < simd_rest; ++i)
+                {
+                    *(res_stepper) = *(fct_stepper);
+                    res_stepper.step_leading();
+                    fct_stepper.step_leading();
+                }
+                strided_assign_detail::idx_tools<layout_type::column_major>::next_idx(idx, max);
+                fct_stepper.to_begin();
+                if (!E1::contiguous_layout)
+                {
+                    res_stepper.to_begin();
+                    for (std::size_t i = cut; i < cut + idx.size(); ++i)
+                    {
+                        fct_stepper.step(i, idx[i]);
+                        res_stepper.step(i, idx[i]);
+                    }
+                }
+                else
+                {
+                    for (std::size_t i = 0; i < idx.size(); ++i)
+                    {
+                        fct_stepper.step(i, idx[i]);
+                    }
+                }
+            }
```

Comment:
This loop is merely the same as the one of row major case (only the template parameter of `idx_tools` and the bounds of the loop in the `!E1::contiguous_layout` branch change), this should be refactored to avoid code duplication.